### ISSUE TITLE
(llava-llama-3-8b-v1_1-hf)  problems of chat

### ISSUE THREAD

**Issue by Yanllan** [InternLM/xtuner#652]
![image](https://github.com/InternLM/xtuner/assets/145842232/c3e1ea26-e1d0-456c-95a7-e87d3d09079c)
When I use the chat template recommended by llava-5-7B (llama3)(https://huggingface.co/xtuner/llava-llama-3-8b-v1_1-hf/discussions/1), the above error occurs. Do I know how to change the code?

**Comment by LZHgrla** [comment]
@Yanllan 

Because there are no tokens that need to be masked out for one-batch generation, there shouldn't be any issues.
If you want to address this warning, you could try passing an attention_mask to `model.generate`, like below code:

```diff
with torch.inference_mode():
    output_ids = model.generate(
        input_ids,
+       attention_mask=torch.ones_like(input_ids).bool(),
        images=image_tensor,
        image_sizes=[image_size],
        do_sample=True if args.temperature > 0 else False,
        temperature=args.temperature,
        max_new_tokens=args.max_new_tokens,
        streamer=streamer,
        stopping_criteria=stop_criteria,
        use_cache=True)
```
